sys/arm64/vmm/vmm_mmu.c

```diff
@@ -309,6 +309,14 @@ vmmpmap_enter(vm_offset_t va, vm_size_t size, vm_paddr_t pa, vm_prot_t prot)
 	if ((prot & VM_PROT_WRITE) == 0) {
 		l3e |= ATTR_S1_AP(ATTR_S1_AP_RO);
 	}
+#if __has_feature(capabilities)
+	if ((prot & VM_PROT_READ_CAP) != 0) {
+		l3e |= ATTR_LC_ENABLED;
+	}
+	if ((prot & VM_PROT_WRITE_CAP) != 0) {
+		l3e |= ATTR_SC;
```

Comment:
Does vmmpmap assume hardware dirty bit management is supported (which isn't true of QEMU)? I don't see even DBM being handled for VM_PROT_WRITE, and a corresponding CDBM here.

Reply:
vmmpmap handles mappings for EL2, which are effectively the same as unmanaged kernel mappings. We don't track their dirty state, so there's no reason to set DBM.